android/src/main/kotlin/io/customer/customer_io/CustomerIoPlugin.kt

```diff
@@ -239,57 +237,38 @@ class CustomerIoPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
          */
     }
 
-    private fun initialize(configData: Map<String, Any>) {
-        // TODO: Fix initialize implementation
-        /*
+    private fun initialize(args: Map<String, Any>): kotlin.Result<Unit> = runCatching {
         val application: Application = context.applicationContext as Application
-        val siteId = configData.getString(Keys.Environment.SITE_ID)
-        val apiKey = configData.getString(Keys.Environment.API_KEY)
-        val region = configData.getProperty<String>(
-            Keys.Environment.REGION
-        )?.takeIfNotBlank()
-        val enableInApp = configData.getProperty<Boolean>(
-            Keys.Environment.ENABLE_IN_APP
-        )
+        val cdpApiKey = requireNotNull(args.getAsTypeOrNull<String>("cdpApiKey")) {
+            "CDP API Key is required to initialize Customer.io"
+        }
 
-        // Checks if SDK was initialized before, which means lifecycle callbacks are already
-        // registered as well
-        val isLifecycleCallbacksRegistered = kotlin.runCatching { CustomerIO.instance() }.isSuccess
+        val logLevelRawValue = args.getAsTypeOrNull<String>("logLevel")
```

Comment:
I used hardcoded strings for keys instead because the constants aren't reused outside this function, so I felt they weren't adding much value. That said, I'm open to moving them back to constants if anyone thinks it would improve clarity or maintainability.